server/re/src/reDataObjOpr.cpp

```diff
@@ -1610,6 +1611,12 @@ msiDataObjChksum( msParam_t *inpParam1, msParam_t *msKeyValStr,
         return rei->status;
     }
 
+    // If the user is an admin, include the admin keyword so that the admin can perform the checksum even if they have
+    // no permissions on the target data object.
+    if (irods::is_privileged_client(*rsComm)) {
+        addKeyVal(&myDataObjInp->condInput, ADMIN_KW, "");
+    }
```

Comment:
> Should we require the policy writer to pass the admin keyword for this to be added?

I contemplated requiring that the `ADMIN_FLAG` be passed (because this MSI does not take keywords). However, I think this would then require the policy implementer to check for the keyword in order to know whether to pass the `ADMIN_FLAG`. I'll toy around with this.

> Are there any downsides to automatically doing this?

The only one I can think of is possibly that it is inconsistent with something like, say, `ichksum`, which does require using an admin option (i.e. `-M`) for objects on which the admin may have no permissions. So, that's a vote for requiring that the `ADMIN_FLAG` be passed.

> Do other microservices do this?

Not that I'm aware of.

Reply:
I can see it going either way, but if I'm the admin writing policy on the server-side, then it's probably better to automatically set the admin flag.

I can't think of a reason not to.

Reply:
I reverted this change (which will obviously get squashed out, but just leaving it for review's sake) and added the `ADMIN_FLAG` support. I also added a section to the test to make sure that regular users get an error when they use the MSI with `ADMIN_FLAG`. Resolving this, but we can continue the conversation in a new thread if needed